components/scream/src/control/surface_coupling.cpp

```diff
@@ -303,6 +307,10 @@ void SurfaceCoupling::do_export ()
     });
   }
 
+  // Deep copy fields to device in case host data was updated
+  // in between registration_end() and do_export()
```

Comment:
I don't understand this: why copying host data to device, given that the exports views will be written on by the next pfor? I assume this has something to do with the init phase, but I can't understand it. Btw, assuming that's the case, shouldn't you do this only if `init_phase` is true?

Reply:
The issue I was having is that, for the test, my cpl data all is -1. If I do not perform a deep copy here, that -1 data gets overwritten to 0 during the deep copy from device to host (only on GPU). So my test was failing because for the initial export, the fields I wasn't exporting I was expecting to be -1, but were 0.

2 things:
1. Will we even care about preserving cpl_data? Or is it fine to send things that are not touched to 0 in the export? I guess this is what we do for fields that aren't relevant to scream..
2. Even if that is not the case, the comment is still incorrect. registration_end() doesn't matter. This is just to preserve any existing data. Should only be done on the init phase. Probably unnecessary on CPU.

Reply:
I could just change the test actually to store the data before registration end...

Reply:
Is this even necessary then? Shouldn't `create_mirror_view()` do a deep copy if the spaces are different?

Reply:
It creates a view on the mirror space, but it doesn't perform the copy. But there is `create_mirror_view_and_copy`, which maybe is what you want?

Reply:
You could move the creation/setup of the SurfaceCoupling in the unit test _inside_ the nruns for loop... Not pretty, maybe, but for such small tests, it's not expensive anyways...